substrate/client/authority-discovery/src/worker.rs

```diff
@@ -550,10 +555,16 @@ where
 			.map_err(|_| Error::ReceivingDhtValueFoundEventWithDifferentKeys)?
 			.ok_or(Error::ReceivingDhtValueFoundEventWithNoRecords)?;
 
-		let authority_id: AuthorityId = self
-			.in_flight_lookups
-			.remove(&remote_key)
-			.ok_or(Error::ReceivingUnexpectedRecord)?;
+		let authority_id: AuthorityId =
+			if let Some(authority_id) = self.in_flight_lookups.remove(&remote_key) {
+				authority_id
+			} else if let Some(authority_id) = self.known_lookups.get(&remote_key) {
+				authority_id.clone()
```

Comment:
It seems this is going to never be called, as in `Discovery` behavior we terminate DHT queries after receiving the first result:
https://github.com/paritytech/polkadot-sdk/blob/e88d1cb79315792a3dbccb6bdef2543093ecaf5b/substrate/client/network/src/discovery.rs#L793-L799

Reply:
Aah, sorry, missed that you also updated the `Discovery` :see_no_evil: